esphome/components/http_request/http_request_idf.cpp

```diff
@@ -19,14 +19,47 @@ namespace http_request {
 
 static const char *const TAG = "http_request.idf";
 
+struct UserData {
+  const std::set<std::string> &collect_header_names;
+  std::map<std::string, std::list<std::string>> response_headers;
+};
+
 void HttpRequestIDF::dump_config() {
   HttpRequestComponent::dump_config();
   ESP_LOGCONFIG(TAG, "  Buffer Size RX: %u", this->buffer_size_rx_);
   ESP_LOGCONFIG(TAG, "  Buffer Size TX: %u", this->buffer_size_tx_);
 }
 
+esp_err_t HttpRequestIDF::http_event_handler(esp_http_client_event_t *evt) {
+  ESP_LOGD(TAG, "Entered http_event_handler");
+  UserData *user_data = (UserData *) evt->user_data;
+
+  switch (evt->event_id) {
+    case HTTP_EVENT_ON_HEADER: {
+      ESP_LOGD(TAG, "Entered HTTP_EVENT_ON_HEADER");
+      ESP_LOGD(TAG, "header name: %s", evt->header_key);
+      const std::string header_name = str_lower_case(evt->header_key);
+      for (const auto &collect_header_name : user_data->collect_header_names) {
+        if (str_equals_case_insensitive(collect_header_name, header_name)) {
+          const std::string header_value = evt->header_value;
+          ESP_LOGD(TAG, "Received response header, name: %s, value: %s", header_name.c_str(), header_value.c_str());
+          user_data->response_headers[header_name].push_back(header_value);
+          break;
+        }
+      }
+      break;
+    }
+    default: {
+      break;
+    }
+  }
+  ESP_LOGD(TAG, "Exit http_event_handler");
```

Comment:
This is logged very often; if you really need to keep it, at least make it verbose.
```suggestion
  ESP_LOGV(TAG, "Exit http_event_handler");
```

Reply:
I removed it